### ISSUE TITLE
Finding the distribution of X/(X+Y) with uniform random variables

### ISSUE THREAD

**Issue by peteroupc** [sympy/sympy#21475]
I am having issues using SymPy to get the distribution of X/(X+Y) when X and Y are uniform.

When expressing it as `P(X<z*(X+Y))`, I get the following error: "'NegativeInfinity' object cannot be interpreted as an integer".  When expressing it as `P(X/(X+Y)<z)`, the expression fails to evaluate --- but it ought to be trivial as can be seen in this [Stack Exchange question](https://math.stackexchange.com/questions/541322/cdf-of-x-xy-where-x-and-y-are-i-i-d-uniform).

Session output:

```

In [494]: x0,x1,y0,y1,z=symbols('x0 x1 y0 y1 z',real=True) 
     ...: xx=Uniform('X',0,1) 
     ...: yy=Uniform('Y',0,1) 
     ...: P(xx<z*(xx+yy)).subs(x0,0).subs(x1,1)                                                                                           
     ...:  
     ...:                                                                                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/usr/local/lib/python3.8/dist-packages/sympy/core/mul.py in _eval_nseries(self, x, n, logx, cdir)
   1824                 n1 = ceiling(n - n0 + m)
-> 1825                 s = t.nseries(x, n=n1, logx=logx, cdir=cdir)
   1826                 ns = s.getn()

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in nseries(self, x, x0, n, dir, logx, cdir)
   3310         else:
-> 3311             return self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   3312 

/usr/local/lib/python3.8/dist-packages/sympy/core/add.py in _eval_nseries(self, x, n, logx, cdir)
    405     def _eval_nseries(self, x, n, logx, cdir=0):
--> 406         terms = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
    407         return self.func(*terms)

/usr/local/lib/python3.8/dist-packages/sympy/core/add.py in <listcomp>(.0)
    405     def _eval_nseries(self, x, n, logx, cdir=0):
--> 406         terms = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
    407         return self.func(*terms)

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in nseries(self, x, x0, n, dir, logx, cdir)
   3310         else:
-> 3311             return self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   3312 

/usr/local/lib/python3.8/dist-packages/sympy/functions/elementary/exponential.py in _eval_nseries(self, x, n, logx, cdir)
    483             nterms = ceiling(n/cf)
--> 484         exp_series = exp(t)._taylor(t, nterms)
    485         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)

/usr/local/lib/python3.8/dist-packages/sympy/functions/elementary/exponential.py in _taylor(self, x, n)
    500         g = None
--> 501         for i in range(n):
    502             g = self.taylor_term(i, self.args[0], g)

TypeError: 'NegativeInfinity' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-494-24d4cf4deac7> in <module>
      2 xx=Uniform('X',0,1)
      3 yy=Uniform('Y',0,1)
----> 4 P(xx<z*(xx+yy)).subs(x0,0).subs(x1,1)

/usr/local/lib/python3.8/dist-packages/sympy/stats/rv.py in probability(condition, given_condition, numsamples, evaluate, **kwargs)
    807     from sympy.stats.symbolic_probability import Probability
    808     if evaluate:
--> 809         return Probability(condition, given_condition).doit(**kwargs)
    810     ### TODO: Remove the user warnings in the future releases
    811     message = ("Since version 1.7, using `evaluate=False` returns `Probability` "

/usr/local/lib/python3.8/dist-packages/sympy/stats/symbolic_probability.py in doit(self, **hints)
    109             return Probability(condition, given_condition)
    110 
--> 111         result = pspace(condition).probability(condition)
    112         if hasattr(result, 'doit') and for_rewrite:
    113             return result.doit()

/usr/local/lib/python3.8/dist-packages/sympy/stats/rv.py in probability(self, condition, **kwargs)
    444         expr = condition.lhs - condition.rhs
    445         rvs = random_symbols(expr)
--> 446         dens = self.compute_density(expr)
    447         if any([pspace(rv).is_Continuous for rv in rvs]):
    448             from sympy.stats.crv import SingleContinuousPSpace

/usr/local/lib/python3.8/dist-packages/sympy/stats/rv.py in compute_density(self, expr, **kwargs)
    471         if any(pspace(rv).is_Continuous for rv in rvs):
    472             z = Dummy('z', real=True)
--> 473             expr = self.compute_expectation(DiracDelta(expr - z),
    474              **kwargs)
    475         else:

/usr/local/lib/python3.8/dist-packages/sympy/stats/rv.py in compute_expectation(self, expr, rvs, evaluate, **kwargs)
    415         rvs = frozenset(rvs)
    416         for space in self.spaces:
--> 417             expr = space.compute_expectation(expr, rvs & space.values, evaluate=False, **kwargs)
    418         if evaluate and hasattr(expr, 'doit'):
    419             return expr.doit(**kwargs)

/usr/local/lib/python3.8/dist-packages/sympy/stats/crv.py in compute_expectation(self, expr, rvs, evaluate, **kwargs)
    658         x = self.value.symbol
    659         try:
--> 660             return self.distribution.expectation(expr, x, evaluate=evaluate, **kwargs)
    661         except PoleError:
    662             return Integral(expr * self.pdf, (x, self.set), **kwargs)

/usr/local/lib/python3.8/dist-packages/sympy/stats/crv_types.py in expectation(self, expr, var, **kwargs)
   3936         from sympy import Max, Min
   3937         kwargs['evaluate'] = True
-> 3938         result = SingleContinuousDistribution.expectation(self, expr, var, **kwargs)
   3939         result = result.subs({Max(self.left, self.right): self.right,
   3940                               Min(self.left, self.right): self.left})

/usr/local/lib/python3.8/dist-packages/sympy/stats/crv.py in expectation(self, expr, var, evaluate, **kwargs)
    426                     return integrate(expr * self.pdf(var), (var, self.set), **kwargs)
    427                 deg = p.degree()
--> 428                 taylor = poly(series(mgf, t, 0, deg + 1).removeO(), t)
    429                 result = 0
    430                 for k in range(deg+1):

/usr/local/lib/python3.8/dist-packages/sympy/series/series.py in series(expr, x, x0, n, dir)
     63     """
     64     expr = sympify(expr)
---> 65     return expr.series(x, x0, n, dir)

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in series(self, x, x0, n, dir, logx, cdir)
   2956             # replace x with an x that has a positive assumption
   2957             xpos = Dummy('x', positive=True, finite=True)
-> 2958             rv = self.subs(x, xpos).series(xpos, x0, n, dir, logx=logx, cdir=cdir)
   2959             if n is None:
   2960                 return (s.subs(xpos, x) for s in rv)

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in series(self, x, x0, n, dir, logx, cdir)
   2963 
   2964         if n is not None:  # nseries handling
-> 2965             s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   2966             o = s1.getO() or S.Zero
   2967             if o:

/usr/local/lib/python3.8/dist-packages/sympy/core/mul.py in _eval_nseries(self, x, n, logx, cdir)
   1831 
   1832         except (ValueError, NotImplementedError, TypeError, AttributeError):
-> 1833             facs = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
   1834             res = powsimp(self.func(*facs).expand(), combine='exp', deep=True)
   1835             if res.has(Order):

/usr/local/lib/python3.8/dist-packages/sympy/core/mul.py in <listcomp>(.0)
   1831 
   1832         except (ValueError, NotImplementedError, TypeError, AttributeError):
-> 1833             facs = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
   1834             res = powsimp(self.func(*facs).expand(), combine='exp', deep=True)
   1835             if res.has(Order):

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in nseries(self, x, x0, n, dir, logx, cdir)
   3309             return self.series(x, x0, n, dir, cdir=cdir)
   3310         else:
-> 3311             return self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   3312 
   3313     def _eval_nseries(self, x, n, logx, cdir):

/usr/local/lib/python3.8/dist-packages/sympy/core/add.py in _eval_nseries(self, x, n, logx, cdir)
    404 
    405     def _eval_nseries(self, x, n, logx, cdir=0):
--> 406         terms = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
    407         return self.func(*terms)
    408 

/usr/local/lib/python3.8/dist-packages/sympy/core/add.py in <listcomp>(.0)
    404 
    405     def _eval_nseries(self, x, n, logx, cdir=0):
--> 406         terms = [t.nseries(x, n=n, logx=logx, cdir=cdir) for t in self.args]
    407         return self.func(*terms)
    408 

/usr/local/lib/python3.8/dist-packages/sympy/core/expr.py in nseries(self, x, x0, n, dir, logx, cdir)
   3309             return self.series(x, x0, n, dir, cdir=cdir)
   3310         else:
-> 3311             return self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
   3312 
   3313     def _eval_nseries(self, x, n, logx, cdir):

/usr/local/lib/python3.8/dist-packages/sympy/functions/elementary/exponential.py in _eval_nseries(self, x, n, logx, cdir)
    482         if cf and cf > 0:
    483             nterms = ceiling(n/cf)
--> 484         exp_series = exp(t)._taylor(t, nterms)
    485         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
    486         if cf and cf > 1:

/usr/local/lib/python3.8/dist-packages/sympy/functions/elementary/exponential.py in _taylor(self, x, n)
    499         l = []
    500         g = None
--> 501         for i in range(n):
    502             g = self.taylor_term(i, self.args[0], g)
    503             g = g.nseries(x, n=n)

TypeError: 'NegativeInfinity' object cannot be interpreted as an integer

In [495]: xx=Uniform('X',0,1) 
     ...: yy=Uniform('Y',0,1) 
     ...: P(xx/(xx+yy)<z).subs(x0,0).subs(x1,1)                                                                                           
     ...:  
     ...:                                                                                                                                 
Out[495]: 
0  1 1                           
⌠  ⌠ ⌠                           
⎮  ⎮ ⎮  ⎛       X      ⎞         
⎮  ⎮ ⎮ δ⎜-z + ───── - z⎟ dX dY dz
⎮  ⎮ ⎮  ⎝     X + Y    ⎠         
⌡  ⌡ ⌡                           
-∞ 0 0                           

In [496]: xx=Uniform('X',x0,x1) 
     ...: yy=Uniform('Y',y0,y1) 
     ...: P(xx/(xx+yy)<z).subs(x0,0).subs(x1,1)                                                                                           
     ...:  
     ...:  
     ...:                                                                                                                                 
Out[496]: 
0                                                                                                                                        
⌠                                                                                                                                        
⎮  ⎛  y₀ 0                           y₀ 1                             0                                 1                            ⎞   
⎮  ⎜  ⌠  ⌠                           ⌠  ⌠                           ⌠ ⌠                               ⌠ ⌠                            ⎟   
⎮  ⎜  ⎮  ⎮  ⎛       X      ⎞         ⎮  ⎮  ⎛       X      ⎞         ⎮ ⎮  ⎛       X       ⎞            ⎮ ⎮  ⎛       X       ⎞         ⎟   
⎮  ⎜  ⎮  ⎮ δ⎜-z + ───── - z⎟ dX dY - ⎮  ⎮ δ⎜-z + ───── - z⎟ dX dY   ⎮ ⎮ δ⎜-z + ────── - z⎟ dX d(y₁) - ⎮ ⎮ δ⎜-z + ────── - z⎟ dX d(y₁)⎟   
⎮  ⎜  ⎮  ⎮  ⎝     X + Y    ⎠         ⎮  ⎮  ⎝     X + Y    ⎠         ⎮ ⎮  ⎝     X + y₁    ⎠            ⎮ ⎮  ⎝     X + y₁    ⎠         ⎟   
⎮  ⎜  ⌡  ⌡                           ⌡  ⌡                           ⌡ ⌡                               ⌡ ⌡                            ⎟   
⎮  ⎜                                                                                                                                 ⎟   
⎮  ⎜- ─────────────────────────────────────────────────────────── + ─────────────────────────────────────────────────────────────────⎟ dz
⎮  ⎝                            y₀ - y₁                                                          y₀ - y₁                             ⎠   
⌡                                                                                                                                        
-∞                        
```

**Comment by joachimlebovits** [comment]
Hi there,

I am also very interested by the answer.
More generally, once we have defined a random vector (let's say the example given in SymPy documentation)

from sympy import exp, pi, Indexed, S
from sympy.stats import density, JointRV
x1, x2 = (Indexed('x', i) for i in (1, 2))
pdf = exp(-x1**2/2 + x1 - x2**2/2 - S(1)/2)/(2*pi)
N1 = JointRV('x', pdf) #Multivariate Normal distribution

How do we compute a probability like P(N1[0]<N1[1])?

In other words, if N1=(N1[0], N1[1]) is multivariate Normal distribution, how do we get access to both N1[0] and N1[1]?

**Comment by prototypevito** [comment]
Update: This no longer gives a TypeError, but instead returns a 0. I'm not able to debug why it is giving a zero though. It'll be great if someone could help. 

>In [494]: x0,x1,y0,y1,z=symbols('x0 x1 y0 y1 z',real=True) 
     ...: xx=Uniform('X',0,1) 
     ...: yy=Uniform('Y',0,1) 
     ...: P(xx<z*(xx+yy)).subs(x0,0).subs(x1,1)
